pyproject.toml

```diff
@@ -37,69 +37,44 @@ exclude = '''
 )/
 '''
 
-
+# Pylint configuration
 [tool.pylint.messages_control]
-# Exceptions suggested from Black: bad-continuation, bad-whitespace
-disable = """
-invalid-name,
-fixme,
-
-bad-continuation,
-
-no-else-return,
-no-else-raise,
-
-no-member,
-
-redefined-variable-type,
-
-abstract-class-instantiated,
-too-many-locals,
-too-few-public-methods,
-too-many-arguments,
-unused-argument,
-missing-module-docstring,
-missing-function-docstring,
-duplicate-code,
-protected-access,
-no-self-use,
-arguments-differ,
-attribute-defined-outside-init,
-too-many-statements,
-too-many-instance-attributes,
-too-complex,
-too-many-lines,
-redefined-builtin,
-abstract-method,
-too-many-branches,
-line-too-long,
-missing-class-docstring,
-unnecessary-pass,
-unused-variable,
-redefined-outer-name,
-function-redefined,
-import-error
-"""
-# Many more `disable`s are defined in `./tox.ini` on a per-module basis!
+disable = [
+    # Exceptions suggested by Black:
+    # https://github.com/psf/black/blob/7f75fe3669ebf0627b1b0476a6d02047e909b959/docs/compatible_configs.md#black-compatible-configurations
+    "bad-continuation",
+    "bad-whitespace",
+    # We allow TODO comments in the following format: `# TODO (#[ISSUE NUMBER]): This needs to be done.`
+    "fixme",
+    # We want to use "mathematical notation" to name some of our variables, e.g. `A` for matrices
+    "invalid-name",
+    # In functions/methods which are part of the public API, we often change the type of argument variables when normalizing inputs for internal processing,
+    # e.g. we might use `pn.utils.as_numpy_scalar` to cast a `float` into a `np.double` or a `np.ndarray` into a `RandomVariable` by using `pn.randvars.asrandvar`.
+    "redefined-variable-type",
+    # Many more `disable`s are defined in `./tox.ini` on a per-module basis!
+]
 
 [tool.pylint.format]
 max-line-length = "88"
 
 [tool.pylint.master]
-extension-pkg-whitelist = "numpy"
-# Extensions that might be of interest in the future:
-# pylint.extensions.check_elif,
-# pylint.extensions.docparams,
-load-plugins = """
-pylint.extensions.docstyle,
-pylint.extensions.mccabe,
-pylint.extensions.redefined_variable_type,
-pylint.extensions.overlapping_exceptions,
-"""
+extension-pkg-whitelist = [
+    "numpy",
+]
+load-plugins = [
+    "pylint.extensions.docstyle",
+    "pylint.extensions.mccabe",
+    "pylint.extensions.redefined_variable_type",
+    "pylint.extensions.overlapping_exceptions",
+    # Extensions that might be of interest in the future:
+    # "pylint.extensions.check_elif",
+    # "pylint.extensions.docparams",
```

Comment:
[minor] now that these lines are touched anyway, is this well-placed here? Would it be better as an issue/discussion instead?

Reply:
I'll just enable the plugins and disable their messages per-package.
That way we can resolve them like any other message.

Reply:
Is it already clear that we want those enabled? Depending on that I agree that you could just enable them, or we might need a discussion on what we actually want to enforce.

Reply:
We certainly want `docparams` enabled and `check_elif` is also a good idea. I'll enable it for now and we can take it out if it causes trouble in the future. Is this fine @pnkraemer @nathanaelbosch?

Reply:
For me this is fine. docparams will detect changes that one often forgets when refactoring (at least I do...). check_elif seems okay to me as well